src/ghci/module_set.rs

```diff
@@ -0,0 +1,94 @@
+use std::borrow::Borrow;
+use std::borrow::Cow;
+use std::cmp::Eq;
+use std::collections::HashSet;
+use std::hash::Hash;
+
+use crate::normal_path::NormalPath;
+
+use super::loaded_module::LoadedModule;
+
+/// A collection of source paths, retaining information about loaded modules in a `ghci`
+/// session.
+#[derive(Debug, Clone, Default, PartialEq, Eq)]
+pub struct ModuleSet {
+    modules: HashSet<LoadedModule>,
+}
```

Comment:
Non-blocking, but I feel like this `struct` wrapper around the `HashSet` doesn't buy you much because most of the methods on `ModuleSet` are trivially wrapping methods on the underlying `HashSet`.  You could probably pass around the `HashSet` directly instead of defining a separate `ModuleSet` `struct`.

Reply:
Yeah, that's true. I like the type safety this gives, though, and I like being able to give semantically descriptive names to the methods. I think I'll keep this for now.